### PR TITLE
feat(exchange.external-contacts): use datagrid component through iceberg

### PR DESCRIPTION
**Pull request by lizardK**
### Description of the Change

- Move from old table component to oui-datagrid
- Improve performances by using iceberg pagination & cache features

### Dependencies

- [x] 2api sws-exchange-externalcontacts

### CODE REVIEW

File: packages/manager/modules/exchange/src/external-contact/external-contact.controller.js [hunk 1]
```diff
@@ -22,33 +22,27 @@ export default class ExchangeTabExternalContactsCtrl {
...
       this.$routerParams.productId,
-      count,
+      pageSize,
       offset,
-      this.filter,
+      criteria.length > 0 ? criteria[0].value : null,
     )
       .then((contacts) => {
-        this.contacts = contacts;
+        this.contacts = contacts.data;
```
Comment (JeremyDec) [inline packages/manager/modules/exchange/src/external-contact/external-contact.controller.js]: You can directly use destructuring for L38 => `.then(({ data, count })`

File: packages/manager/modules/exchange/src/external-contact/external-contact.html [hunk 2]
```diff
@@ -3,220 +3,117 @@
...
-                        </tr>
-                    </tbody>
-
-                    <tbody
-                        data-ng-if="!ctrl.contactsLoading && ctrl.contacts.list.results.length === 0"
+                            data-translate="exchange_tab_EXTERNAL_CONTACTS_table_actions_menu_modify"
+                        ></span>
+                    </oui-action-menu-item>
+                    <oui-action-menu-item
+                        data-on-click="setAction('exchange/external-contact/remove/external-contact-remove', $row.externalEmailAddress)"
```
Comment (JeremyDec) [inline packages/manager/modules/exchange/src/external-contact/external-contact.html]: In order to harmonize with component properties ? :)
```suggestion
                       on-click="setAction('exchange/external-contact/remove/external-contact-remove', $row.externalEmailAddress)"
```

File: packages/manager/modules/exchange/src/external-contact/external-contact.html [hunk 3]
```diff
@@ -3,220 +3,117 @@
...
-                                data-ng-disabled="ctrl.contactsLoading"
-                            >
-                                <span
-                                    class="fa fa-times"
-                                    aria-hidden="true"
-                                ></span>
-                            </button>
-                        </div>
+                <oui-column
+                    title="'exchange_tab_EXTERNAL_CONTACTS_table_headers_externalEmailAddress' | translate"
```
Comment (FredericEspiau) [inline packages/manager/modules/exchange/src/external-contact/external-contact.html]:
```suggestion
                    data-title="'exchange_tab_EXTERNAL_CONTACTS_table_headers_externalEmailAddress' | translate"
```

`title` is a `html5` attribute 🙂

File: packages/manager/modules/exchange/src/external-contact/external-contact.html [hunk 4]
```diff
@@ -3,220 +3,117 @@
...
+                            data-translate="exchange_tab_EXTERNAL_CONTACTS_table_actions_menu_delete"
+                        ></span>
+                    </oui-action-menu-item>
+                </oui-action-menu>
 
-                    <tbody data-ng-if="!ctrl.contactsLoading">
-                        <tr
-                            data-ng-repeat="element in ctrl.contacts.list.results track by $index"
+                <extra-top>
+                    <oui-action-menu text="Actions" align="start">
```
Comment (FredericEspiau) [inline packages/manager/modules/exchange/src/external-contact/external-contact.html]: Please translate `Actions`

File: packages/manager/modules/exchange/src/external-contact/external-contact.html [hunk 5]
```diff
@@ -3,220 +3,117 @@
     data-ng-controller="ExchangeTabExternalContactsCtrl as ctrl"
 >
     <div class="row">
-        <div class="col-md-9">
-            <form
-                class="form-inline d-md-flex justify-content-md-end mb-3"
-                name="searchExternalContactForm"
+        <div class="col-md-12">
```
Comment (antleblanc) [inline packages/manager/modules/exchange/src/external-contact/external-contact.html]: I think you can then rely on a single `<div>` element (`.row > .col-XX-12` === `<div>`).

File: packages/manager/modules/exchange/src/external-contact/external-contact.html [hunk 6]
```diff
@@ -3,220 +3,117 @@
...
+                    title="'exchange_tab_EXTERNAL_CONTACTS_table_headers_displayName' | translate"
+                    property="displayName"
+                >
+                </oui-column>
+                <oui-column
+                    title="'exchange_tab_EXTERNAL_CONTACTS_table_headers_creationDate' | translate"
+                    property="creationDate"
+                    type="date"
+                >
+                    <span>{{ $value | date:'mediumDate' }}</span>
```
Comment (antleblanc) [inline packages/manager/modules/exchange/src/external-contact/external-contact.html]: Prefer using the `ngBind` directive over `{{`.

File: packages/manager/modules/exchange/src/external-contact/external-contact.html [hunk 7]
```diff
@@ -3,220 +3,117 @@
...
+                </oui-column>
+                <oui-column
+                    title="'exchange_tab_EXTERNAL_CONTACTS_table_headers_hiddenFromGAL' | translate"
+                    property="hiddenFromGAL"
+                >
+                    <span
+                        class="oui-icon"
+                        aria-hidden="true"
+                        data-ng-class="{'oui-icon-success text-success': !$value.hiddenFromGAL, 'oui-icon-error text-danger': $value.hiddenFromGAL}"
```
Comment (antleblanc) [inline packages/manager/modules/exchange/src/external-contact/external-contact.html]: Both helpers `text-success` and `text-danger` still needed?

File: packages/manager/modules/exchange/src/external-contact/external-contact.html [hunk 8]
```diff
@@ -3,220 +3,117 @@
...
+                    </oui-action-menu-item>
+                </oui-action-menu>
 
-                    <tbody data-ng-if="!ctrl.contactsLoading">
-                        <tr
-                            data-ng-repeat="element in ctrl.contacts.list.results track by $index"
+                <extra-top>
+                    <oui-action-menu text="Actions" align="start">
+                        <oui-action-menu-item
+                            text="{{'exchange_tab_EXTERNAL_CONTACTS_configuration_contact_add_title_button' | translate}}"
```
Comment (antleblanc) [inline packages/manager/modules/exchange/src/external-contact/external-contact.html]: Could you please avoid using the deprecated `text` attribute?

Prefer using a transclude.

```html
<span data-translate="exchange_tab_EXTERNAL_CONTACTS_configuration_contact_add_title_button"></span>
```

See: https://ovh-ux.github.io/ovh-ui-kit-documentation/#!/oui-angular/action-menu#deprecated

File: packages/manager/modules/exchange/src/external-contact/external-contact.html [hunk 9]
```diff
@@ -2,221 +2,120 @@
     class="container-fluid px-0"
     data-ng-controller="ExchangeTabExternalContactsCtrl as ctrl"
 >
-    <div class="row">
-        <div class="col-md-9">
-            <form
-                class="form-inline d-md-flex justify-content-md-end mb-3"
-                name="searchExternalContactForm"
+    <div>
```
Comment (JeremyDec) [inline packages/manager/modules/exchange/src/external-contact/external-contact.html]: You can now : 
```suggestion
```

File: packages/manager/modules/exchange/src/external-contact/external-contact.html [hunk 10]
```diff
@@ -2,221 +2,120 @@
...
+        <oui-datagrid rows-loader="ctrl.loadContacts($config)" page-size="10">
+            <oui-column
+                data-title="'exchange_tab_EXTERNAL_CONTACTS_table_headers_externalEmailAddress' | translate"
```
Comment (marie-j) [inline packages/manager/modules/exchange/src/external-contact/external-contact.html]:
```suggestion
                data-title=:: "'exchange_tab_EXTERNAL_CONTACTS_table_headers_externalEmailAddress' | translate"
```

File: packages/manager/modules/exchange/src/external-contact/external-contact.html [hunk 11]
```diff
@@ -2,221 +2,120 @@
...
+                property="creationDate"
+                type="date"
+            >
+                <span data-ng-bind="$value | date:'mediumDate'"></span>
+            </oui-column>
+            <oui-column
+                data-title="'exchange_tab_EXTERNAL_CONTACTS_table_headers_hiddenFromGAL' | translate"
+                property="hiddenFromGAL"
+            >
+                <span
```
Comment (marie-j) [inline packages/manager/modules/exchange/src/external-contact/external-contact.html]: No alternative content for `sr-only` ?

Comment (antleblanc) [reply packages/manager/modules/exchange/src/external-contact/external-contact.html]: I will remove the `aria-hidden` attribute (not needed).

Refer to: https://ovh-ux.github.io/ovh-ui-kit-documentation/#!/oui-angular/tooltip#accessibility

**Pull request merged**